crates/uv/tests/tool_install.rs

```diff
@@ -2998,12 +3001,105 @@ fn tool_install_at_latest_upgrade() {
         [tool]
         requirements = [{ name = "black" }]
         entrypoints = [
-            { name = "black", install-path = "[TEMP_DIR]/bin/black" },
-            { name = "blackd", install-path = "[TEMP_DIR]/bin/blackd" },
+            { name = "black", install-path = "[TEMP_DIR]/bin/black", from = "black" },
+            { name = "blackd", install-path = "[TEMP_DIR]/bin/blackd", from = "black" },
+        ]
+
+        [tool.options]
+        exclude-newer = "2024-03-25T00:00:00Z"
+        "###);
+    });
+}
+
+/// Test installing a tool together with some additional entrypoints
+/// from other packages.
+#[test]
+fn tool_install_additional_entrypoints() {
+    let context = TestContext::new("3.12")
+        .with_filtered_counts()
+        .with_filtered_exe_suffix();
+    let tool_dir = context.temp_dir.child("tools");
+    let bin_dir = context.temp_dir.child("bin");
+
+    uv_snapshot!(context.filters(), context.tool_install()
+        .arg("--i-want-ponies")
+        .arg("ansible-core")
+        .arg("--i-want-ponies")
+        .arg("black")
+        .arg("ansible==9.3.0")
+        .env("UV_TOOL_DIR", tool_dir.as_os_str())
+        .env("XDG_BIN_HOME", bin_dir.as_os_str())
+        .env("PATH", bin_dir.as_os_str()), @r###"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+
+    ----- stderr -----
+    Resolved [N] packages in [TIME]
+    Prepared [N] packages in [TIME]
+    Installed [N] packages in [TIME]
+     + ansible==9.3.0
+     + ansible-core==2.16.4
+     + black==24.3.0
+     + cffi==1.16.0
+     + click==8.1.7
+     + cryptography==42.0.5
+     + jinja2==3.1.3
+     + markupsafe==2.1.5
+     + mypy-extensions==1.0.0
+     + packaging==24.0
+     + pathspec==0.12.1
+     + platformdirs==4.2.0
+     + pycparser==2.21
+     + pyyaml==6.0.1
+     + resolvelib==1.0.1
+    Installed 11 executables from `ansible-core`: ansible, ansible-config, ansible-connection, ansible-console, ansible-doc, ansible-galaxy, ansible-inventory, ansible-playbook, ansible-pull, ansible-test, ansible-vault
```

Comment:
There is a funny issue here, it seems that on Windows only there is a different sorting for the strings `ansible` and `ansible-<whatever>`:
```diff
- Installed 11 executables from `ansible-core`: ansible, ansible-config, ansible-connection, ansible-console, ansible-doc, ansible-galaxy, ansible-inventory, ansible-playbook, ansible-pull, ansible-test, ansible-vault
+ Installed 11 executables from `ansible-core`: ansible-config, ansible-connection, ansible-console, ansible-doc, ansible-galaxy, ansible-inventory, ansible-playbook, ansible-pull, ansible-test, ansible-vault, ansible
```

Reply:
That's horrible :(

Reply:
I had to abuse CI a bit to get to the bottom of this (sorry if I flooded you with notifications).
It turns out there is a `.exe` suffix (filtered out in test output) and thus `ansible < ansible-config(.exe) < ansible.exe` (due to the ASCII order of `-` vs `.`).
The same suffix is also present in the receipt, making its ordering OS-dependant.